src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,366 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Component, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import * as chartjs from 'chart.js';
+import { mixin } from 'sql/base/common/objects';
+import { Disposable } from 'vs/base/common/lifecycle';
+import * as azdata from 'azdata';
+import { generateUuid } from 'vs/base/common/uuid';
+
+export interface BarDataSet {
+	label: string;
+	data: number[];
+	backgroundColor?: string[];
+	borderColor?: string[];
+}
+
+@Component({
+	selector: 'chart-component',
+	templateUrl: decodeURI(require.toUrl('./chart.component.html'))
+})
+export class Chart<TChartType extends azdata.ChartType, TData extends azdata.ChartData<TChartType>, TOptions extends azdata.ChartOptions<TChartType>> extends Disposable {
+	private _type: TChartType;
+	private _data: chartjs.ChartData;
+
+	private chart: chartjs.Chart;
+	private canvas: HTMLCanvasElement;
+	private chartCanvasId: string;
+
+	/**
+	 * Options in the form that Chart.js accepts
+	 */
+	private _options: chartjs.ChartOptions = {
+		events: ['click', 'keyup'],
+		responsive: true,
+		maintainAspectRatio: false
+	};
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef
+	) {
+		chartjs.Chart.register(...chartjs.registerables);
+		super();
+
+		this.chartCanvasId = 'chart' + generateUuid();
+	}
+
+	ngAfterViewInit(): void {
+
+	}
+
+	/**
+	 * Setter function for chart type
+	 */
+	public set type(val: any) {
```

Comment:
This should be `ChartType` or at least string, right? Although _type is currently `TChartType` which doesn't seem right if you intend to have this be updated after initial creation (and if not then should you even have this setter?)

Reply:
Ah, it should be `TChartType`.  Chart's can't change types, but `modelComponents\chart.component.ts` still needs to be able to set this when all the initial property values are getting set, right?